### PR TITLE
issues(4918): fix resolve scoped service for Elsa.MongoDb.Modules

### PR DESCRIPTION
**Pull request by bbenameur**
Releated to this issues: https://github.com/elsa-workflows/elsa-core/issues/4918

### CODE REVIEW

**Comment by sfmskywalker** [comment]
Your PR makes sense, and I think we can get by with a single scope that we create first and then pass down to the individual "create indices" methods. What do you think?

**Comment by bbenameur** [comment]
I agree this is the correct way

File: src/modules/Elsa.MongoDb/Modules/Identity/CreateIndices.cs [hunk 1]
```diff
@@ -8,9 +8,9 @@ namespace Elsa.MongoDb.Modules.Identity;
 
 internal class CreateIndices : IHostedService
 {
-    private readonly IServiceProvider _serviceProvider;
+    private readonly IServiceScope _serviceScope;
```
Comment (sfmskywalker) [inline src/modules/Elsa.MongoDb/Modules/Identity/CreateIndices.cs:11]: Although I am not 100% sure, but if this hosted service remains in memory, so does this child service scope. Therefore, perhaps it is better to not store the child scope asa private field, but rather as a temporary local variable that gets disposed when done with.

E.g.

```csharp
private readonly IServiceProvider _serviceProvider;

    public CreateIndices(IServiceProvider serviceProvider) => _serviceProvider = serviceProvider;

    public Task StartAsync(CancellationToken cancellationToken)
    {
       using var scope = _serviceProvider.CreateScope();
        return Task.WhenAll(
            CreateWorkflowDefinitionIndices(scope, cancellationToken),
            CreateWorkflowInstanceIndices(scope, cancellationToken));
    }
```

**Review by sfmskywalker — APPROVED** [review]
Awesome 👍🏻

**Pull request merged**